docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -0,0 +1,51 @@
+# JSON_DIAGNOSTIC_POSITIONS
+
+```cpp
+#define JSON_DIAGNOSTIC_POSITIONS /* value */
+```
+
+This macro enables position diagnostics for generated JSON objects.
+
+When enabled, two new properties: `start_pos()` and `end_pos()` are added to `nlohmann::json` objects and fields. `start_pos()`
```

Comment:
A nit with the overall explanation, we don't define what end/start means, would probably be useful, from a documentation perspective, to explain what constitutes a start/end for an object vs field